src/Adapter/Language/CommandHandler/AbstractLanguageHandler.php

```diff
@@ -144,4 +146,24 @@ protected function getLegacyLanguageObject(LanguageId $languageId)
 
         return $language;
     }
+
+    /**
+     * @param Language $language
+     */
+    protected function assertIsNotDefaultLanguage(Language $language)
+    {
+        if ($language->id === (int) Configuration::get('PS_LANG_DEFAULT')) {
+            throw new DefaultLanguageException(sprintf('Default language "%s" cannot be deleted', $language->iso_code), DefaultLanguageException::CANNOT_DELETE_ERROR);
+        }
+    }
+
+    /**
+     * @param Language $language
+     */
+    protected function assertIsNotInUseLanguage(Language $language)
+    {
+        if ($language->id === (int) Context::getContext()->language->id) {
+            throw new DefaultLanguageException(sprintf('Used language "%s" cannot be deleted', $language->iso_code), DefaultLanguageException::CANNOT_DELETE_IN_USE_ERROR);
+        }
+    }
```

Comment:
Renamed to avoid a conflict with https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Language/CommandHandler/ToggleLanguageStatusHandler.php#L63
 In any case I prepare another PR to improve the Toogles classes too.

Reply:
Perfect.